modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1362,7 +1362,13 @@ void GDScriptAnalyzer::resolve_for(GDScriptParser::ForNode *p_for) {
 					if (all_is_constant) {
 						switch (args.size()) {
 							case 1:
-								reduced = args[0];
+								if (args[0].get_type() == Variant::FLOAT) {
+									Callable::CallError temp;
+									const Variant *construct_arg = &args[0];
+									Variant::construct(Variant::INT, reduced, &construct_arg, 1, temp);
```

Comment:
There should probably be error checking here since the float could be something like `INF`

Reply:
As discussed in the meeting, having this check for INF and NaN would be nice.

Reply:
Just found out that there is no check for normal `range` yet.
```gdscript
var x = range(INF) # INF gets converted to int (becomes INT_MIN) and used directly. Returns `[]`
```
Also if it should be checked, should the other 2 parameters be checked as well?
During runtime for non constant `range`?

I started to think maybe the checks can be added with another PR... (as the Variant::construct here never "fails", it's as safe as before)